File: Week-01/01-Baseball Game/Mamdouh.cpp
        public class Solution
        {
            public int CalPoints(string[] operations)
            {
                Stack<int> stack = new();
                int result = 0;
                int number = 0;
                foreach(var op in operations)
                {
                    if (op == "+")
                    {
                        int first = stack.Pop();
                        int second = stack.Peek();
                        stack.Push(first);
                        stack.Push(first + second);
                        result += (first + second);
                    }
                    else if (op == "C")
                    {
                        result -= stack.Pop();
                        //stack.Pop();
                    }
                    else if (op == "D")
                    {
                        number = stack.Peek() * 2;
                        result += number;
                        stack.Push(number);

                    }

                    else
                    {
                        number = int.Parse(op);
                        stack.Push(number);
                        result += number;
                    }
                }

                return result;
            }
        }

File: Week-01/10-Daily Temperatures/Mamdouh.cs
Author : Ahmed Mamdouh
Time : O(n), Space : O(n)

public class Solution
 {
     public int[] DailyTemperatures(int[] temperatures)
     {

         int[] result = new int[temperatures.Length];
         Stack<(int, int)> minMonoStack = new();
         for(int i=0;i < temperatures.Length; i++)
         {
             while (minMonoStack.Count > 0 && (temperatures[i] > minMonoStack.Peek().Item1))
             {
                 int numberIndex = minMonoStack.Peek().Item2;
                 result[numberIndex] = i - numberIndex;
                 minMonoStack.Pop();
             }
             minMonoStack.Push((temperatures[i], i));
         }

         while (minMonoStack.Count > 0)
         {
             result[minMonoStack.Peek().Item2] = 0;
             minMonoStack.Pop();
         }
         return result;
     }
 }
